### PR TITLE
DEVPROD-4511 Remove db logging from honeycomb

### PR DESCRIPTION
**Pull request by bynn**
DEVPROD-4511

### Description
don't log db statements in honeycomb because it could leak secrets

What removing [WithCommandAttributeDisabled()](https://github.com/mongodb/anser/blob/c62f11870fd43cfbdcc66dfc9db68fcdce047d4c/apm/otel_monitor.go#L86-L93) does is set our monitor to the default behavior of no longer adding MongoDB commands as attributes

According to @ybrill , that was the only reason we use a copy of the open telemetry library in anser and we could go back to using the [original](https://github.com/open-telemetry/opentelemetry-go-contrib/tree/main/instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo) if we don't plan on using this feature again. I don't believe that is necessary at the moment so no follow up tickets have been made. Similarly, no ticket has been created to re-enable this span after sanitizing our commands because it may not be necessary.

### Testing
no longer logs db.statement in staging 
image

### CODE REVIEW

**Review by ablack12 — COMMENTED** [review]
I don't know enough about honeycomb; would you mind putting some words about how this prevents logging db.statement? (Also, I think a comment would be good, so we don't later think this change was just an oversight)

**Comment by bynn** [comment]
Added some comments and updated the description

**Comment by khelif96** [comment]
Sorry for the drive by review. But is it possible to reconsider this or keep it enabled for traces stemming from a graphql query? Having the db statement is invaluable for debugging performance issues or debugging problematic queries that can't be easily replicated. I was under the impression that refinery instance already redacts secrets from db queries? Cc @ybrill

**Comment by ybrill** [comment]
The application omits `db.statement` for certain collections and then refinery does some rudimentary scrubbing.

File: environment.go [hunk 1]
```diff
@@ -368,7 +368,8 @@ func (e *envState) initDB(ctx context.Context, settings DBSettings, tracer trace
 	opts := options.Client().ApplyURI(settings.Url).SetWriteConcern(settings.WriteConcernSettings.Resolve()).
 		SetReadConcern(settings.ReadConcernSettings.Resolve()).
 		SetConnectTimeout(5 * time.Second).
-		SetMonitor(apm.NewMonitor(apm.WithCommandAttributeDisabled(false), apm.WithCommandAttributeTransformer(redactSensitiveCollections)))
+		// Removing WithCommandAttributeDisabled(false) to disable db command monitoring.
```
[the remark this answers is not in the document]
Comment (bynn) [reply environment.go:371]: I wonder if db.statement without any values would still be useful though? 
I'm not super sure on what you mean by "would have to be in the collector" could I ask you to explain that part

[the remark this answers is not in the document]
Comment (ybrill) [reply environment.go:371]: That's true: the query plan could be different with different values.

Sorry, for not being clear in the first place 😅. In the application, when we're making the span for the database call the only thing we know about our lineage is id of our parent and the overall trace id. So we don't have a way to know whether the trace we're part of is a graphql one. On the other hand, if we wanted to drop the attribute once the span arrives at the collector we could have the application add a flag attribute to all the child spans of a graphql request and when we see a db span with that flag then we'll know to keep the `db.statement`.

**Comment by bynn** [comment]
Upon discussing this ticket further with @ybrill and seeing that @khelif96 both have recently used this span, I decided to close this ticket as won't do. The data sanitization work done previously seems to mitigate most security risks. 

@ablack12 let me know if you have any opinions

**Pull request closed without merging**